datafusion/core/tests/sqllogictests/test_files/functions.slt

```diff
@@ -458,3 +458,26 @@ SELECT v1, v2, ROWNUMBER() OVER(ORDER BY v1) from test;
 
 statement ok
 drop table test
+
+# Scalar function struct
+statement ok
+create table simple_struct_test (
```

Comment:
Could you please also add an example test where the column names don't happen to line up with the column names?

Like 
```sql
create table test(x boolean) as values (true);
select struct(x) from test
```
?

I expect the output is named `c1` even though the column is named `x`

To make the output struct named `x` I think would be a significant change to the type resolution code as it would need to get a `Schema` rather than a `[DataType]`

Reply:
thanks for your help.
i have updated the current test case: 

```
create table simple_struct_test (
    c1 boolean,
    c2 INT,
    c3 FLOAT,
    c4 DOUBLE,
    a VARCHAR,
    b TEXT
) as select *
from (values
  (true, 1,3.1,3.14,'str','text')
);

```

it's should cover the scenario you mentioned.